### ISSUE TITLE
[Plugin] Segfault on loading custom image

### ISSUE THREAD

**Issue by Crazycolbster** [OpenRCT2/OpenRCT2#21632]
### Operating System

Linux

### OpenRCT2 build

OpenRCT2, v0.4.9 (, DEBUG)

### Base game

RollerCoaster Tycoon 2

### Describe the issue

While figuring out how to add pictures to a plugin, I caused a segfault. The game produced the following error:
`An assertion failed, please report this to the OpenRCT2 developers.
Version: OpenRCT2, v0.4.9 (, DEBUG)
Location: ReadPng:129
openrct2 $ openrct2: ./src/openrct2/core/Guard.cpp:92: void Guard::Assert_VA(bool, const char*, __va_list_tag*): Assertion `false' failed.
`

### Steps to reproduce

Run initCustomSprites() from the attached file.

### Attachments

[custom_images.ts.zip](https://github.com/OpenRCT2/OpenRCT2/files/14629778/custom_images.ts.zip)

**Comment by Crazycolbster** [comment]
Checking in on this a couple years later, and it does appear this still crashes the game, this time with some more context. Running initCustomSprites() throws the following error before crashing the game:
```
An assertion failed, please report this to the OpenRCT2 developers.
Version: OpenRCT2, v0.5.3
Location: ReadPng:134
openrct2 $ terminate called after throwing an instance of 'std::invalid_argument'
                                                                                   what():  Only images 300x300 or less are supported.
                                           Aborted (core dumped)
```

**Comment by James103** [comment]
The script you're trying to run embeds an image as base-64 data within the script file; I was able to decode it and get its resolution: 2034 × 2112 pixels. Can you please try downscaling it by a factor of 8 before loading it again so it will fit within the stated bounds here?
https://github.com/OpenRCT2/OpenRCT2/blob/cef7dd164ba8394ae0b7b43f961ded962e06ddb8/src/openrct2/drawing/ImageImporter.cpp#L32-L35

**Comment by Die4Ever** [comment]
I still feel like plugins should not be capable of segfaulting the parent process

**Comment by Crazycolbster** [comment]
That's the big thing to my understanding. I already have a functional image library in my plugin, but I've been told before that the end goal for the plugin engine is that plugins should never be able to crash the game.

**Comment by Basssiiie** [comment]
That is correct. Thank you for your report. If you find more, please keep reporting them!

The preferable fix would be that it doesn't crash the game.
